src/samplers/mongodbSampler.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  Sampler,
+  ParentBasedSampler,
+  SamplingResult,
+  SamplingDecision,
+} from '@opentelemetry/sdk-trace-base';
+import { Context, Link, Attributes, SpanKind } from '@opentelemetry/api';
+import { logger } from '../logging';
+
+export class MongodbSampler implements Sampler {
+  /* eslint-disable @typescript-eslint/no-unused-vars */
+  shouldSample(
+    context: Context,
+    traceId: string,
+    spanName: string,
+    spanKind: SpanKind,
+    attributes: Attributes,
+    links: Link[]
+  ): SamplingResult {
+    // Note, there is probably a bug in opentelemetry api, making mongoSampler always receives attributes array empty.
+    // This makes it impossible to filter based on db.system and db.operation attributes. Filter based on spanName only.
+    // Opentemetry version upgrade might fix this issue.
+    // https://lumigo.atlassian.net/browse/RD-14250
+    let decision = SamplingDecision.RECORD_AND_SAMPLED;
```

Comment:
I'm usually refraining from using `let`, as IMO it's less easy to follow and les explicit. WDYT about:
```ts
if (spanKind === SpanKind.CLIENT && matchMongoIsMaster(spanName, dbSystem, dbOperation)) {
      logger.debug(
        `Drop span ${spanName} with db.system: ${dbSystem} and db.operation: ${dbOperation}, because LUMIGO_REDUCED_MONGO_INSTRUMENTATION is enabled`
      );
      return SamplingDecision.NOT_RECORD;
    }
```

and then the fallback `return` becomes:
```ts
return { decision: SamplingDecision.RECORD_AND_SAMPLED; };
```

Reply:
agree